### PR TITLE
Add unit_tests and drive by assorted

### PR DESCRIPTION
**Pull request by gnuoy**
* Add unit tests
* Fix path to repo in layers.yaml
* Ignore unit_tests etc in layers.yaml
* Make set_state calls consistently from conversation
* Setup travis (interim step until this interface is in OpenStack
  gerrit)

### CODE REVIEW

File: unit_tests/test_provides.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+    def tearDown(self):
+        self.cr = None
+        for k, v in self._patches.items():
+            v.stop()
+            setattr(self, k, None)
+        self._patches = None
+        self._patches_start = None
+
+    def patch_kr(self, attr, return_value=None):
```
Comment (fnordahl) [inline unit_tests/test_provides.py]: nit: the origin of this is probably the keystone interface unit tests and I think the abbreviation `kr` refers to "keystone relation".  Would it be appropriate to either change the abbreviation to match constructs in this interface or pick a generic term we can re-use across interface unit tests?

**Review by fnordahl — COMMENTED** [review]
Generally ok with this but I do have a comment, see in-line.

File: unit_tests/test_provides.py (hunk 1 above)
Comment (gnuoy) [reply unit_tests/test_provides.py]: Good point. Should be fixed now.

**Review by fnordahl — APPROVED** [review]
LGTM, thanks!

**Pull request merged**